code/datums/status_effects/debuffs/fire_stacks.dm

```diff
@@ -244,8 +244,10 @@
 	if(!silent)
 		owner.visible_message(span_warning("[owner] catches fire!"), span_userdanger("You're set on fire!"))
 
-	if(firelight_type)
-		firelight_ref = WEAKREF(new firelight_type(owner))
+	if(moblight_type)
+		if(moblight)
+			qdel(moblight)
+		moblight = new moblight_type(owner)
```

Comment:
should `mob_light()` support passing in a type

Reply:
I don't  know, should it? Should I give it that argument? I can do so if you think it's a good idea.